### PR TITLE
Updated ResultsParser and added Project789 and dependentTestGenerator…

### PR DESCRIPTION
**Pull request by achiu17**
Added Project789 in data and dependentTestGenerator.java in Main to generate figures 7,8, and 9

### CODE REVIEW

**Comment by winglam** [comment]
Please resolve your conflicts when making a pull request (before and after you work, git pull from my repository). At the current state not only can I not merge your pull request but it is also out of date, making it difficult to know what you actually changed.

**Comment by achiu17** [comment]
Resolved conflicts

**Comment by winglam** [comment]
Is this ready for my review now? I see that the dependentTestGenerator and some other stuff I mentioned in the progress email is still not addressed.

**Comment by achiu17** [comment]
Which progress email are you referring to? I don't think I received a progress email regarding any changes needed to be made for the dependentTestGenerator.

**Comment by achiu17** [comment]
Made super/subclasses for FigureGenerator and Project

File: dt-impact-tracer/src/edu/washington/cs/dt/impact/Main/NumDependentTestsFigureGenerator.java [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+
+        File directory = new File(directoryName);
+        File[] fList = directory.listFiles();
+        // create a list of project Objects that each have a diff project name
+        int size = 5;
+        List<Project> proj_orig_arrayList = new ArrayList<Project>(size);
+        List<Project> proj_auto_arrayList = new ArrayList<Project>(size);
+
+        for (File file : fList) {
+            if (file.isFile()) {
```
Comment (winglam) [inline dt-impact-tracer/src/edu/washington/cs/dt/impact/Main/NumDependentTestsFigureGenerator.java:157]: You should be able to refactor this code into a method and have both this class and the EnhancedResultsFigureGenerator call such method.

File: dt-impact-tracer/src/edu/washington/cs/dt/impact/Main/ResultsParser.java [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (winglam) [inline dt-impact-tracer/src/edu/washington/cs/dt/impact/Main/ResultsParser.java:1]: Please delete this clas, dependentTestGenerator and Project789 if they are no longer needed.

**Comment by achiu17** [comment]
Modified Fig 18 to display S1-S6 and fixed phantomref errors

**Comment by winglam** [comment]
LGTM!

**Pull request merged**